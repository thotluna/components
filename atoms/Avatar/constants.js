export const options = {
  size: ['sm', 'md', 'lg'],
}
