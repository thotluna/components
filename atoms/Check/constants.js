export const options = {
  colors: ['primary', 'secondary'],
  sizes: ['sm', 'md', 'lg'],
}
