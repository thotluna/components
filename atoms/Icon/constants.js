import React from 'react'
export const options = {
  colors: ['base', 'primary', 'error'],
  sizes: ['sm', 'md', 'lg'],
  types: [
    'alert',
    'check',
    'circle',
    'close',
    'down-chevron',
    'drag',
    'left-chevron',
    'menu',
    'plus',
    'save',
    'trash',
  ],
}

export const getIcon = (type) => {
  switch (type) {
    case 'save':
      return (
        <svg
          height={21}
          viewBox="0 0 21 21"
          width={21}
          xmlns="http://www.w3.org/2000/svg"
          stroke="currentColor"
        >
          <g
            fill="none"
            fillRule="evenodd"
            strokeLinecap="round"
            strokeLinejoin="round"
          >
            <path d="M6.5 10.5l4 4.232 4-4.191M10.5 3.5v11M4.5 17.5h12" />
          </g>
        </svg>
      )
    case 'alert':
      return (
        <svg
          height={21}
          viewBox="0 0 21 21"
          width={21}
          xmlns="http://www.w3.org/2000/svg"
          stroke="currentColor"
        >
          <g fill="none" fillRule="evenodd" transform="translate(1 1)">
            <path
              d="M9.5.5l9 16H.5zM9.5 10.5v-5"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
            <circle cx={9.5} cy={13.5} fill="currentColor" r={1} />
          </g>
        </svg>
      )
    case 'circle':
      return (
        <svg
          height={21}
          viewBox="0 0 21 21"
          width={21}
          xmlns="http://www.w3.org/2000/svg"
          stroke="currentColor"
        >
          <g
            fill="none"
            fillRule="evenodd"
            strokeLinecap="round"
            strokeLinejoin="round"
          >
            <path d="M14.857 3.79a8 8 0 102.852 3.24M6.5 9.5l3" />
          </g>
        </svg>
      )
    case 'check':
      return (
        <svg
          height={21}
          viewBox="0 0 21 21"
          width={21}
          xmlns="http://www.w3.org/2000/svg"
          stroke="currentColor"
        >
          <g
            fill="none"
            fillRule="evenodd"
            strokeLinecap="round"
            strokeLinejoin="round"
          >
            <path d="M14.857 3.79a8 8 0 102.852 3.24M6.5 9.5l3 3 8-8" />
          </g>
        </svg>
      )
    case 'close':
      return (
        <svg
          height={21}
          viewBox="0 0 21 21"
          width={21}
          xmlns="http://www.w3.org/2000/svg"
          stroke="currentColor"
        >
          <g
            fill="none"
            fillRule="evenodd"
            strokeLinecap="round"
            strokeLinejoin="round"
          >
            <path d="M7.5 7.5l6 6M13.5 7.5l-6 6" />
          </g>
        </svg>
      )
    case 'menu':
      return (
        <svg
          height={21}
          viewBox="0 0 21 21"
          width={21}
          xmlns="http://www.w3.org/2000/svg"
          fill="currentColor"
        >
          <g fill="none" fillRule="evenodd" transform="translate(2 2)">
            <path d="M8.5 9.5c.5 0 1-.5 1-1s-.5-1-1-1-.999.5-.999 1 .499 1 .999 1zm-4 0c.5 0 1-.5 1-1s-.5-1-1-1-.999.5-.999 1 .499 1 .999 1zm8 0c.5 0 1-.5 1-1s-.5-1-1-1-.999.5-.999 1 .499 1 .999 1z" />
          </g>
        </svg>
      )
    case 'left-chevron':
      return (
        <svg
          height={21}
          viewBox="0 0 21 21"
          width={21}
          xmlns="http://www.w3.org/2000/svg"
          stroke="currentColor"
        >
          <path
            d="M11.5 14.5l-4-4 4-4"
            fill="none"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      )
    case 'down-chevron':
      return (
        <svg
          height="21"
          viewBox="0 0 21 21"
          width="21"
          xmlns="http://www.w3.org/2000/svg"
          stroke="currentColor"
        >
          <path
            d="m8.5.5-4 4-4-4"
            fill="none"
            strokeLinecap="round"
            strokeLinejoin="round"
            transform="translate(6 8)"
          />
        </svg>
      )
    case 'trash':
      return (
        <svg
          height={21}
          viewBox="0 0 21 21"
          width={21}
          xmlns="http://www.w3.org/2000/svg"
          stroke="currentColor"
        >
          <g
            fill="none"
            fillRule="evenodd"
            strokeLinecap="round"
            strokeLinejoin="round"
          >
            <path d="M5.5 4.5h10v12a2 2 0 01-2 2h-6a2 2 0 01-2-2zm5-2a2 2 0 011.995 1.85l.005.15h-4a2 2 0 012-2zM3.5 4.5h14M8.5 7.5v8M12.5 7.5v8" />
          </g>
        </svg>
      )
    case 'drag':
      return (
        <svg
          height={21}
          viewBox="0 0 21 21"
          width={21}
          xmlns="http://www.w3.org/2000/svg"
        >
          <g fill="currentColor" fillRule="evenodd">
            <path d="M7 5h2v2H7zM12 5h2v2h-2zM7 9h2v2H7zM12 9h2v2h-2zM7 13h2v2H7zM12 13h2v2h-2z" />
          </g>
        </svg>
      )
    case 'plus':
      return (
        <svg
          height={21}
          viewBox="0 0 21 21"
          width={21}
          xmlns="http://www.w3.org/2000/svg"
          stroke="currentColor"
        >
          <g
            fill="none"
            fillRule="evenodd"
            strokeLinecap="round"
            strokeLinejoin="round"
          >
            <path d="M10 19c4.438 0 8-3.526 8-7.964C18 6.598 14.438 3 10 3c-4.438 0-8 3.598-8 8.036S5.562 19 10 19zM6 11h8M10 15.056V7z" />
          </g>
        </svg>
      )

    default:
      break
  }
}
