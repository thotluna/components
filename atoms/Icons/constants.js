import React from 'react'
export const options = {
  colors: ['base', 'primary', 'secondary', 'inverted', 'sun'],
  size: ['sm', 'md', 'lg'],
  type: [
    'chevron-left',
    'chevron-down',
    'check',
    'drag',
    'moon',
    'sun',
    'trash',
    'close',
  ],
}

export const getIcon = (icon) => {
  switch (icon) {
    case 'chevron-left':
      return (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          width="100%"
          height="100%"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            d="M15 19l-7-7 7-7"
          />
        </svg>
      )
    case 'chevron-down':
      return (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          width="100%"
          height="100%"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            d="M19 9l-7 7-7-7"
          />
        </svg>
      )
    case 'check':
      return (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          width="100%"
          height="100%"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            d="M5 13l4 4L19 7"
          />
        </svg>
      )
    case 'trash':
      return (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          width="100%"
          height="100%"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
          />
        </svg>
      )
    case 'drag':
      return (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          width="80%"
          height="80%"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            d="M7 11.5V14m0-2.5v-6a1.5 1.5 0 113 0m-3 6a1.5 1.5 0 00-3 0v2a7.5 7.5 0 0015 0v-5a1.5 1.5 0 00-3 0m-6-3V11m0-5.5v-1a1.5 1.5 0 013 0v1m0 0V11m0-5.5a1.5 1.5 0 013 0v3m0 0V11"
          />
        </svg>
      )
    case 'moon':
      return (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          width="100%"
          height="100%"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1"
            d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"
          />
        </svg>
      )
    case 'sun':
      return (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          width="100%"
          height="100%"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z"
          />
        </svg>
      )
    case 'close':
      return (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 21 21"
          stroke="currentColor"
          width="100%"
          height="100%"
        >
          <g
            fill="none"
            fillRule="evenodd"
            strokeLinecap="round"
            strokeLinejoin="round"
          >
            <path d="M15.5 15.5l-10-10zM15.5 5.5l-10 10" />
          </g>
        </svg>
      )
  }
}
