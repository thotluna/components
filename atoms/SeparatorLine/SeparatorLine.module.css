.container-separator {
  position: relative;
  width: 100%;
  margin: 5px 0;
}

.separator-line {
  width: 100%;
}

.title {
  position: absolute;
  top: -13px;
  left: 2rem;
  padding: 0.5rem;
  background: var(--color-background);
  color: var(--color-font-muted);
}

.color-base {
  border: none;
  background-color: var(--color-font-base);
}

.color-inverted {
  border: none;
  background-color: var(--color-primary-inverted);
}

.color-primary {
  border: none;
  background-color: var(--color-primary);
}

.color-secondary {
  border: none;
  background-color: var(--color-secondary);
}

.color-tertiary {
  border: none;
  background-color: var(--color-font-highlight);
}

.size-xs {
  height: 1px;
}

.size-sm {
  height: 2px;
}

.size-md {
  height: 3px;
}

.size-lg {
  height: 5px;
}
