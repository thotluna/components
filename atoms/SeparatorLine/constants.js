export const options = {
  colors: ['base', 'primary', 'secondary'],
  sizes: ['xs', 'sm', 'md', 'lg'],
}
