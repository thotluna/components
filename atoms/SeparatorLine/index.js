export { default, SeparatorLine } from './SeparatorLine'
export { options } from './constants'
export { default as styles } from './SeparatorLine.module.css'
