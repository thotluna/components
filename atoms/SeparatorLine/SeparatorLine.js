import React from 'react'
import PropTypes from 'prop-types'
import styles from './SeparatorLine.module.css'
import { options } from './constants'
import withStyles from '../../hocs/withStyles'

import Heading from '../Headings'

export const SeparatorLine = ({ title, isShowTitle, color, getStyles }) => {
  return (
    <div className={getStyles('container-separator', ['color'])}>
      <hr
        title={title}
        className={getStyles('separator-line', ['color', 'size'])}
      />
      {isShowTitle && (
        <div className={getStyles('title')}>
          <Heading color={color}>{title}</Heading>
        </div>
      )}
    </div>
  )
}

SeparatorLine.propTypes = {
  getStyles: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  isShowTitle: PropTypes.bool,
  color: PropTypes.oneOf(options.colors),
  size: PropTypes.oneOf(options.sizes),
}

SeparatorLine.defaultProps = {
  size: 'sm',
  color: 'primary',
  isShowTitle: false,
}

export default withStyles(styles)(SeparatorLine)
