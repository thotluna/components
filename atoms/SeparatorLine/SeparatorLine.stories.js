import { SeparatorLine } from './SeparatorLine'
import styles from './SeparatorLine.module.css'
import { options } from './constants'

import {
  getTemplate,
  getListTemplate,
  getOptionsArgTypes,
} from '../../helpers/storybook'

const Template = getTemplate(SeparatorLine, styles)
const ListTemplate = getListTemplate(SeparatorLine, styles)

export default {
  title: 'Atoms/SeparatorLine',
  component: SeparatorLine,
  args: {
    title: 'separator',
    size: 'sm',
  },
  argTypes: {
    color: getOptionsArgTypes(options.colors),
    size: getOptionsArgTypes(options.sizes),
  },
}

export const Default = Template.bind({})
Default.args = { title: 'separator' }

export const Colors = ListTemplate.bind({})
Colors.args = {
  items: options.colors.map((color) => ({ title: 'separator', color })),
}

export const Size = ListTemplate.bind({})
Size.args = {
  items: options.sizes.map((size) => ({ title: 'separator', size })),
}

export const TitelColors = ListTemplate.bind({})
TitelColors.args = {
  items: options.colors.map((color) => ({
    title: 'separator',
    isShowTitle: true,
    color,
  })),
}

export const TitelSizes = ListTemplate.bind({})
TitelSizes.args = {
  items: options.sizes.map((size) => ({
    title: 'separator',
    isShowTitle: true,
    size,
  })),
}
