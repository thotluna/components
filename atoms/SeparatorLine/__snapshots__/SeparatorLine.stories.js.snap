// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/SeparatorLine Colors 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="container-separator color-base"
  >
    <hr
      className="separator-line color-base size-sm"
      title={null}
    />
  </div>
  <div
    className="container-separator color-primary"
  >
    <hr
      className="separator-line color-primary size-sm"
      title={null}
    />
  </div>
  <div
    className="container-separator color-secondary"
  >
    <hr
      className="separator-line color-secondary size-sm"
      title={null}
    />
  </div>
  <div
    className="container-separator color-tertiary"
  >
    <hr
      className="separator-line color-tertiary size-sm"
      title={null}
    />
  </div>
  <div
    className="container-separator color-inverted"
  >
    <hr
      className="separator-line color-inverted size-sm"
      title={null}
    />
  </div>
</div>
`;

exports[`Storyshots Atoms/SeparatorLine Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="container-separator color-primary"
  >
    <hr
      className="separator-line color-primary size-sm"
      title={null}
    />
  </div>
</div>
`;

exports[`Storyshots Atoms/SeparatorLine Size 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="container-separator color-primary"
  >
    <hr
      className="separator-line color-primary size-xs"
      title={null}
    />
  </div>
  <div
    className="container-separator color-primary"
  >
    <hr
      className="separator-line color-primary size-sm"
      title={null}
    />
  </div>
  <div
    className="container-separator color-primary"
  >
    <hr
      className="separator-line color-primary size-md"
      title={null}
    />
  </div>
  <div
    className="container-separator color-primary"
  >
    <hr
      className="separator-line color-primary size-lg"
      title={null}
    />
  </div>
</div>
`;

exports[`Storyshots Atoms/SeparatorLine Titel Colors 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="container-separator color-base"
  >
    <hr
      className="separator-line color-base size-sm"
      title="separator"
    />
    <div
      className="title"
    >
      <header
        className="headings color-base size-md weight-bold"
      >
        separator
      </header>
    </div>
  </div>
  <div
    className="container-separator color-primary"
  >
    <hr
      className="separator-line color-primary size-sm"
      title="separator"
    />
    <div
      className="title"
    >
      <header
        className="headings color-primary size-md weight-bold"
      >
        separator
      </header>
    </div>
  </div>
  <div
    className="container-separator color-secondary"
  >
    <hr
      className="separator-line color-secondary size-sm"
      title="separator"
    />
    <div
      className="title"
    >
      <header
        className="headings color-secondary size-md weight-bold"
      >
        separator
      </header>
    </div>
  </div>
  <div
    className="container-separator color-tertiary"
  >
    <hr
      className="separator-line color-tertiary size-sm"
      title="separator"
    />
    <div
      className="title"
    >
      <header
        className="headings color-tertiary size-md weight-bold"
      >
        separator
      </header>
    </div>
  </div>
  <div
    className="container-separator color-inverted"
  >
    <hr
      className="separator-line color-inverted size-sm"
      title="separator"
    />
    <div
      className="title"
    >
      <header
        className="headings color-inverted size-md weight-bold"
      >
        separator
      </header>
    </div>
  </div>
</div>
`;

exports[`Storyshots Atoms/SeparatorLine Titel Sizes 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="container-separator color-primary"
  >
    <hr
      className="separator-line color-primary size-xs"
      title="separator"
    />
    <div
      className="title"
    >
      <header
        className="headings color-primary size-md weight-bold"
      >
        separator
      </header>
    </div>
  </div>
  <div
    className="container-separator color-primary"
  >
    <hr
      className="separator-line color-primary size-sm"
      title="separator"
    />
    <div
      className="title"
    >
      <header
        className="headings color-primary size-md weight-bold"
      >
        separator
      </header>
    </div>
  </div>
  <div
    className="container-separator color-primary"
  >
    <hr
      className="separator-line color-primary size-md"
      title="separator"
    />
    <div
      className="title"
    >
      <header
        className="headings color-primary size-md weight-bold"
      >
        separator
      </header>
    </div>
  </div>
  <div
    className="container-separator color-primary"
  >
    <hr
      className="separator-line color-primary size-lg"
      title="separator"
    />
    <div
      className="title"
    >
      <header
        className="headings color-primary size-md weight-bold"
      >
        separator
      </header>
    </div>
  </div>
</div>
`;
